src/zenml/cli/stack.py

```diff
@@ -564,33 +564,45 @@ def describe_stack(stack_name: Optional[str]) -> None:
 
 @stack.command("delete")
 @click.argument("stack_name", type=str)
-def delete_stack(stack_name: str) -> None:
+@click.option("--yes", "-y", is_flag=True, required=False)
+def delete_stack(stack_name: str, yes: bool = False) -> None:
     """Delete a stack."""
     cli_utils.print_active_profile()
+    if yes:
+        confirmation = True
+    else:
+        confirmation = cli_utils.confirmation(
+            f"This will delete stack '{stack_name}' from your repository. \n"
+            "Are you sure you want to proceed?"
+        )
 
-    with console.status(f"Deleting stack '{stack_name}'...\n"):
-        cfg = GlobalConfiguration()
-        repo = Repository()
-
-        if cfg.active_stack_name == stack_name:
-            cli_utils.error(
-                f"Stack {stack_name} cannot be deleted while it is globally "
-                f"active. Please choose a different active global stack first "
-                f"by running 'zenml stack set --global STACK'."
-            )
-            return
-
-        if repo.active_stack_name == stack_name:
-            cli_utils.error(
-                f"Stack {stack_name} cannot be deleted while it is "
-                f"active. Please choose a different active stack first by "
-                f"running 'zenml stack set STACK'."
-            )
-            return
+    if confirmation:
```

Comment:
Instead of indenting lot's of the code here (which often makes me wonder what the else clause is doing), we could raise the error message early here and leave the remaining code outside of any if-condition:
```python
if not confirmation:
    cli_utils.declare("Stack deletion cancelled.")

# remaining code as before
with console.status(f"Deleting stack '{stack_name}'...\n"):
    ...
```

Reply:
Thank you yes, I like this a lot more. I was trying to figure out how to best handle this. Your way is much nicer.